src/Compiler/Checking/MethodCalls.fs

```diff
@@ -279,21 +282,33 @@ let rec AdjustRequiredTypeForTypeDirectedConversions (infoReader: InfoReader) ad
 
     // Adhoc int32 --> int64
     elif g.langVersion.SupportsFeature LanguageFeature.AdditionalTypeDirectedConversions && typeEquiv g g.int64_ty reqdTy && typeEquiv g g.int32_ty actualTy then 
-       g.int32_ty, TypeDirectedConversionUsed.Yes(warn TypeDirectedConversion.BuiltIn), None
+        g.int32_ty, TypeDirectedConversionUsed.Yes(warn TypeDirectedConversion.BuiltIn, false), None
 
     // Adhoc int32 --> nativeint
     elif g.langVersion.SupportsFeature LanguageFeature.AdditionalTypeDirectedConversions && typeEquiv g g.nativeint_ty reqdTy && typeEquiv g g.int32_ty actualTy then 
-       g.int32_ty, TypeDirectedConversionUsed.Yes(warn TypeDirectedConversion.BuiltIn), None
+        g.int32_ty, TypeDirectedConversionUsed.Yes(warn TypeDirectedConversion.BuiltIn, false), None
 
     // Adhoc int32 --> float64
     elif g.langVersion.SupportsFeature LanguageFeature.AdditionalTypeDirectedConversions && typeEquiv g g.float_ty reqdTy && typeEquiv g g.int32_ty actualTy then 
-       g.int32_ty, TypeDirectedConversionUsed.Yes(warn TypeDirectedConversion.BuiltIn), None
+        g.int32_ty, TypeDirectedConversionUsed.Yes(warn TypeDirectedConversion.BuiltIn, false), None
 
+    elif g.langVersion.SupportsFeature LanguageFeature.NullableOptionalInterop && isMethodArg && isNullableTy g reqdTy && not (isNullableTy g actualTy) then 
```

Comment:
Could you explain this additional rule a little more for me.   I see it's introducing the two-step conversion. 

Also why only `isMethodArg` here?  In the other rules, `isMethodArg` is only used to determine whether we give a warning.

Reply:
This rule is mostly transplanting checks that existed on the relevant call sites to make sure it only hits in the same cases as the callers used to. Important however for the behavior from both RFCs is to keep this conversion limited to method args only.

I wrote about that in the PR
>Type directed conversions for System.Nullable outside method args still fall through to op_Implicit, meaning that we still error on let x: Nullable<float> = 1: int as expected.

We don't want to enable double conversions for these cases as it was never specced to be allowed.

Beyond that though, the two step conversion was always there and called out in FS-1093 (starting from here https://github.com/fsharp/fslang-design/blob/main/FSharp-6.0/FS-1093-additional-conversions.md#interaction-with-optional-arguments). 

It was just split between ad-hoc checks outside that called into the TDC with the nullable underlying type. (see https://github.com/dotnet/fsharp/pull/13673/files#diff-33f19e1cff6e1df09e36df423722449340f685c5c08f041543334c173cc259f8L355)

Reply:
OK, I understand, thank you